src/main/java/io/dockstore/provision/DOSPlugin.java

```diff
@@ -49,24 +52,65 @@ public void stop() {
     @Extension
     public static class DOSPreProvision implements PreProvisionInterface {
 
-        DOSPluginUtil dosPluginUtil = new DOSPluginUtil();
         static final Set<String> SCHEME = new HashSet<>(Lists.newArrayList("dos"));
+        static final String SCHEME_PREFERENCE = "scheme-preference";
+        static final int GET_SCHEME = 0;
+        private Map<String, String> config;
+
+        DOSPluginUtil dosPluginUtil = new DOSPluginUtil();
+        List<String> preferredSchemes = new ArrayList<>();
+
+        @Override
+        public void setConfiguration(Map<String, String> map) {
+            this.config = map;
+
+            // Parse the preferred schemes into a list by splitting on commas and extra spaces, and remove all empty strings from the list
+            if (this.config.containsKey(SCHEME_PREFERENCE)) {
+
+                // Arrays.asList returns a fixed-sized array, so wrapping it as an ArrayList object allows list modification
+                this.preferredSchemes = new ArrayList<>(Arrays.asList(this.config.get(SCHEME_PREFERENCE).trim().split(",\\s*")));
```

Comment:
This is clever, but would probably be more readable by using standard libraries for properties. 
In particular, consider looking at the list handling section of https://commons.apache.org/proper/commons-configuration/userguide/howto_properties.html 

I believe that we're using https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/HierarchicalConfiguration.html for properties in this file which should inherit methods like getStringArray